x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/hooks/use_expanded_card.test.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useExpandedCard } from './use_expanded_card';
+import { HEIGHT_ANIMATION_DURATION } from '../onboarding_card_panel.styles';
+import type { OnboardingCardId } from '../../../constants';
+import { mockReportCardOpen } from '../../__mocks__/onboarding_context_mocks';
+import { waitFor } from '@testing-library/react';
+
+const scrollTimeout = HEIGHT_ANIMATION_DURATION + 50;
+
+const mockSetStorageExpandedCardId = jest.fn();
+const mockUseStoredExpandedCardId = jest.fn(() => [null, mockSetStorageExpandedCardId]);
+jest.mock('../../../hooks/use_stored_state', () => ({
+  useStoredExpandedCardId: () => mockUseStoredExpandedCardId(),
+}));
+
+jest.mock('react-router-dom', () => ({
+  useLocation: () => ({ hash: '#card-1', pathname: '/test' }),
```

Comment:
`...jest.requireActual` please

I've personally run into this exact mock blowing up tests and taking me hours to trace this line in another project :)